File: AlgorithmsProblemSolvingPython/leetcode/Arrays/bestBuySellStock.py
import math

def maxProfit(prices):
    min_price = math.inf
    max_profit = 0

    for i in range(len(prices)):
        if prices[i] < min_price:
            min_price = prices[i]
        if prices[i] - min_price > max_profit:
            max_profit = prices[i] - min_price

    return max_profit

print(maxProfit([7,1,5,3,6,4]))

File: AlgorithmsProblemSolvingJava/src/io/programminglife/leetcode/Arrays/BestBuySellStock.java
package io.programminglife.leetcode.Arrays;

public class BestBuySellStock {
    
    public static int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxDiff = 0;

        for(int i = 0; i < prices.length; i++) {
            if(prices[i] < minPrice) {
                minPrice = prices[i];
            }
            if(prices[i] - minPrice > maxDiff) {
                maxDiff = prices[i] - minPrice;
            }
        }
        
        return maxDiff;
    }

    public static void main(String[] args) {
        int[] prices = {7,1,5,3,6,4};

        System.out.println(maxProfit(prices));
    }
    
}